src/merge.ts

```diff
@@ -1,24 +1,112 @@
+      if (!schemas.find(s => isEqual(s, schema))) {
+        return schemas.concat(schema);
+      }
+      return schemas;
+    },
+    isExclusivelyAnyOfSchema(schema1) ? schema1.anyOf : [schema1],
+  );
+
+  return schemas.length === 1 ? schemas[0] : { anyOf: schemas as JSONSchema7[] };
+}
+
+function mergeParams<T extends IHttpParam>(params1: T[], params2: T[]): T[] {
+  const params1OnlyAndCommon = params1.map(p1 => {
+    const p2 = params2.find(p2 => p2.name.toLowerCase() === p1.name.toLowerCase());
+
+    if (p2) {
+      return {
+        ...p2,
+        ...p1,
+        required: p1.required && p2.required,
+        schema: p1.schema && p2.schema ? mergeSchemas(p1.schema, p2.schema) : p1.schema ? p1.schema : p2.schema,
+      };
+    } else {
+      return {
+        ...p1,
+        required: false,
+      };
+    }
+  });
+
+  const params2Only = params2
+    .filter(p2 => !params1.find(p1 => p1.name.toLowerCase() === p2.name.toLowerCase()))
+    .map(h2 => ({ ...h2, required: false }));
+
+  return [...params1OnlyAndCommon, ...params2Only];
+}
 
 const mergeContents = mergeLists<IMediaTypeContent[]>(
   (c1, c2) => c1.mediaType.toLowerCase() === c2.mediaType.toLowerCase(),
-  c1 => c1, // ignore content #2 if mediaTypes is equal
+  (c1, c2) => {
+    return {
+      mediaType: c1.mediaType,
+      schema: c1.schema && c2.schema ? mergeSchemas(c1.schema, c2.schema) : c1.schema ? c1.schema : c2.schema,
+      examples: mergeContentExamples([c1.examples, c2.examples]),
+      encodings: mergeContentEncodings([c1.encodings, c2.encodings]),
+    };
+  },
 );
 
+function mergeContentExamples(exampleLists: Array<IMediaTypeContent['examples']>) {
+  return exampleLists.reduce<Array<INodeExample | INodeExternalExample> | undefined>((merged, examples) => {
+    if (!examples) return merged;
+
+    const arr = (Array.isArray(examples) ? examples : [examples]).filter(
+      ex => !(merged || []).find(me => isEqual(me, ex)),
+    );
+
+    return merged ? merged.concat(arr) : arr;
+  }, undefined);
+}
+
+function mergeContentEncodings(encodingLists: Array<IMediaTypeContent['encodings']>) {
+  return encodingLists.reduce<IMediaTypeContent['encodings']>((merged, encodings) => {
+    if (!encodings) return merged;
+
+    const arr = encodings.filter(enc => !(merged || []).find(me => isEqual(me, enc)));
+
+    return merged ? merged.concat(arr) : arr;
+  }, undefined);
+}
+
 export const mergeResponses = mergeLists<IHttpOperation['responses']>(
   (r1, r2) => r1.code === r2.code,
   (r1, r2) => ({
     ...r1,
-    headers: mergeHeaders(r1.headers || [], r2.headers || []),
+    headers: mergeParams(r1.headers || [], r2.headers || []),
     contents: mergeContents(r1.contents || [], r2.contents || []),
   }),
 );
 
+function mergeRequestBodies(b1: IHttpOperationRequestBody, b2: IHttpOperationRequestBody): IHttpOperationRequestBody {
+  return {
+    description: [b1.description, b2.description].filter(_ => _).join('; ') || undefined,
```

Comment:
`[].join('; ')` returns empty string. Empty string is falsy. As a result, we get `undefined` instead of empty string. Which makes more logical sense IMO than defaulting to `''`.